### PR TITLE
skills menu tgui

### PR DESCRIPTION
**Pull request by stanalbatross**
# About the pull request

converted the skills menu to tgui because all the remaining nanouis are scary

normal version :
image

admin version :
image

(and yes I added a rights check in the ui act so you CANNOT EXPLOIT IT)

# Changelog

:cl:
ui: converted the check skills menu to TGUI
admin: changed the edit skills vv dropdown from a clunky series of input lists to the new check skills menu TGUI
/:cl:

### CODE REVIEW

**Review by harryob — APPROVED** [review]

File: tgui/packages/tgui/interfaces/SkillsMenu.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  );
+};
+
+const SkillsView = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { skillset_name, skills, owner, admin } = data;
+  return skills.map((skill, index) => (
+    <Fragment key={index}>
+      <ProgressBar value={skill.level / skill.maxlevel}>
+        {capitalize(skill.name)}: {skill.level}/{skill.maxlevel}
```
Comment (harryob) [inline tgui/packages/tgui/interfaces/SkillsMenu.js:46]:
```suggestion
        {capitalizeAll(skill.name)}: {skill.level}/{skill.maxlevel}
```
this prevents JTAC being converted to Jtac and makes sure Illegal Technology etc is properly uppercased

File: tgui/packages/tgui/interfaces/SkillsMenu.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { capitalize } from 'common/string';
```
Comment (harryob) [inline tgui/packages/tgui/interfaces/SkillsMenu.js:1]:
```suggestion
import { capitalizeAll } from 'common/string';
```

File: code/datums/skills.dm [hunk 3]
```diff
@@ -91,47 +117,69 @@
...
 
 /datum/skill/vehicles
 	skill_name = SKILL_VEHICLE
+	readable_skill_name = SKILL_VEHICLE
 	skill_level = SKILL_VEHICLE_DEFAULT
+	max_skill_level = SKILL_VEHICLE_MAX
 
 /datum/skill/jtac
 	skill_name = SKILL_JTAC
+	readable_skill_name = SKILL_JTAC
```
Comment (harryob) [inline code/datums/skills.dm:156]:
```suggestion
	readable_skill_name = "JTAC"
```

File: code/modules/mob/living/carbon/human/powers/check_skills.dm [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/mob/living/carbon/human/verb/checkSkills()
+	set name = "Check Skills"
+	set category = "IC"
+	set src = usr
+
+	src.skills.tgui_interact(src)
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/powers/check_skills.dm:6]: https://github.com/tgstation/tgstation/pull/72499

asdad; laksf;la ks;dl aks;dl aksd 

```suggestion
	skills.tgui_interact(src)
```

File: code/modules/mob/living/carbon/human/powers/check_skills.dm [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		data["skills"] = null
+		return data // they have no skills =  show the null skills menu
+
+	var/list/skills_data_list = list()
+
+	for(var/skillname as anything in skills)
+		var/datum/skill/skilldatum = get_skill(skillname)
+
+		skills_data_list += list(list(
+			"name" = skilldatum.readable_skill_name,
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/powers/check_skills.dm:42]:
```suggestion
			"name" = skilldatum.readable_skill_name ? skilldatum.readable_skill_name : skilldatum.skill_name
```
then we don't need to double up when readable_skill_name is just the same as skill_name, as in 90% of cases there

**Review by harryob — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/SkillsMenu.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        minValue={0}
+        maxValue={skill.maxlevel}
+        onChange={(e, value) =>
+          act('set_skill', {
+            skill: skill.realname,
+            level: value,
+            oldlevel: skill.level,
+          })
+        }>
+        {capitalize(skill.name)}: {skill.level}/{skill.maxlevel}
```
Comment (stanalbatross) [inline tgui/packages/tgui/interfaces/SkillsMenu.js:70]:
```suggestion
        {capitalizeAll(skill.name)}: {skill.level}/{skill.maxlevel}
```

File: code/modules/mob/living/carbon/human/powers/check_skills.dm [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		data["skills"] = null
+		return data // they have no skills =  show the null skills menu
+
+	var/list/skills_data_list = list()
+
+	for(var/skillname as anything in skills)
+		var/datum/skill/skilldatum = get_skill(skillname)
+
+		skills_data_list += list(list(
+			"name" = skilldatum.readable_skill_name ? skilldatum.readable_skill_name : skilldatum.skill_name
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/human/powers/check_skills.dm:40]:
```suggestion
			"name" = skilldatum.readable_skill_name ? skilldatum.readable_skill_name : skilldatum.skill_name,
```

File: code/datums/skills.dm [hunk 8]
```diff
@@ -18,35 +24,51 @@
...
 /datum/skill/melee_weapons
 	skill_name = SKILL_MELEE_WEAPONS
+	readable_skill_name = "melee weapons"
 	skill_level = SKILL_MELEE_DEFAULT
+	max_skill_level = SKILL_MELEE_MAX
 
 /datum/skill/firearms
 	skill_name = SKILL_FIREARMS
+	readable_skill_name = SKILL_FIREARMS
 	skill_level = SKILL_FIREARMS_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:40]:
```suggestion
	skill_name = SKILL_FIREARMS
	skill_level = SKILL_FIREARMS_DEFAULT
```

File: code/datums/skills.dm [hunk 9]
```diff
@@ -18,35 +24,51 @@
...
 /datum/skill/spec_weapons
 	skill_name = SKILL_SPEC_WEAPONS
+	readable_skill_name = "specialist weapons"
 	skill_level = SKILL_SPEC_DEFAULT
+	max_skill_level = SKILL_SPEC_ALL
 
 /datum/skill/endurance
 	skill_name = SKILL_ENDURANCE
+	readable_skill_name = SKILL_ENDURANCE
 	skill_level = SKILL_ENDURANCE_WEAK
```
Comment (harryob) [inline code/datums/skills.dm:52]:
```suggestion
	skill_name = SKILL_ENDURANCE
	skill_level = SKILL_ENDURANCE_WEAK
```

File: code/datums/skills.dm [hunk 10]
```diff
@@ -18,35 +24,51 @@
...
+	max_skill_level = SKILL_ENDURANCE_MAX
 
 /datum/skill/engineer
 	skill_name = SKILL_ENGINEER
+	readable_skill_name = SKILL_ENGINEER
 	skill_level = SKILL_ENGINEER_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:58]:
```suggestion
	skill_name = SKILL_ENGINEER
	skill_level = SKILL_ENGINEER_DEFAULT
```

File: code/datums/skills.dm [hunk 11]
```diff
@@ -18,35 +24,51 @@
...
+	max_skill_level = SKILL_ENGINEER_MAX
 
 /datum/skill/construction
 	skill_name = SKILL_CONSTRUCTION
+	readable_skill_name = SKILL_CONSTRUCTION
 	skill_level = SKILL_CONSTRUCTION_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:64]:
```suggestion
	skill_name = SKILL_CONSTRUCTION
	skill_level = SKILL_CONSTRUCTION_DEFAULT
```

File: code/datums/skills.dm [hunk 12]
```diff
@@ -18,35 +24,51 @@
...
+	max_skill_level = SKILL_CONSTRUCTION_MAX
 
 /datum/skill/leadership
 	skill_name = SKILL_LEADERSHIP
+	readable_skill_name = SKILL_LEADERSHIP
 	skill_level = SKILL_LEAD_NOVICE
```
Comment (harryob) [inline code/datums/skills.dm:70]:
```suggestion
	skill_name = SKILL_LEADERSHIP
	skill_level = SKILL_LEAD_NOVICE
```

File: code/datums/skills.dm [hunk 13]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/intel
 	skill_name = SKILL_INTEL
+	readable_skill_name = SKILL_INTEL
 	skill_level = SKILL_INTEL_NOVICE
+	max_skill_level = SKILL_INTEL_MAX
 
 /datum/skill/domestic
 	skill_name = SKILL_DOMESTIC
+	readable_skill_name = SKILL_DOMESTIC
 	skill_level = SKILL_DOMESTIC_NONE
```
Comment (harryob) [inline code/datums/skills.dm:175]:
```suggestion
	skill_name = SKILL_DOMESTIC
	skill_level = SKILL_DOMESTIC_NONE
```

File: code/datums/skills.dm [hunk 14]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/execution
 	skill_name = SKILL_EXECUTION
+	readable_skill_name = SKILL_EXECUTION
 	skill_level = SKILL_EXECUTION_DEFAULT
+	max_skill_level = SKILL_EXECUTION_MAX
 
 /datum/skill/intel
 	skill_name = SKILL_INTEL
+	readable_skill_name = SKILL_INTEL
 	skill_level = SKILL_INTEL_NOVICE
```
Comment (harryob) [inline code/datums/skills.dm:169]:
```suggestion
	skill_name = SKILL_INTEL
	skill_level = SKILL_INTEL_NOVICE
```

File: code/datums/skills.dm [hunk 15]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/jtac
 	skill_name = SKILL_JTAC
+	readable_skill_name = "JTAC"
 	skill_level = SKILL_JTAC_NOVICE
+	max_skill_level = SKILL_JTAC_MAX
 
 /datum/skill/execution
 	skill_name = SKILL_EXECUTION
+	readable_skill_name = SKILL_EXECUTION
 	skill_level = SKILL_EXECUTION_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:163]:
```suggestion
	skill_name = SKILL_EXECUTION
	skill_level = SKILL_EXECUTION_DEFAULT
```

File: code/datums/skills.dm [hunk 16]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/powerloader
 	skill_name = SKILL_POWERLOADER
+	readable_skill_name = SKILL_POWERLOADER
 	skill_level = SKILL_POWERLOADER_DEFAULT
+	max_skill_level = SKILL_POWERLOADER_MAX
 
 /datum/skill/vehicles
 	skill_name = SKILL_VEHICLE
+	readable_skill_name = SKILL_VEHICLE
 	skill_level = SKILL_VEHICLE_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:151]:
```suggestion
	skill_name = SKILL_VEHICLE
	skill_level = SKILL_VEHICLE_DEFAULT
```

File: code/datums/skills.dm [hunk 17]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/police
 	skill_name = SKILL_POLICE
+	readable_skill_name = SKILL_POLICE
 	skill_level = SKILL_POLICE_DEFAULT
+	max_skill_level = SKILL_POLICE_MAX
 
 /datum/skill/powerloader
 	skill_name = SKILL_POWERLOADER
+	readable_skill_name = SKILL_POWERLOADER
 	skill_level = SKILL_POWERLOADER_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:145]:
```suggestion
	skill_name = SKILL_POWERLOADER
	skill_level = SKILL_POWERLOADER_DEFAULT
```

File: code/datums/skills.dm [hunk 18]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/pilot
 	skill_name = SKILL_PILOT
+	readable_skill_name = SKILL_PILOT
 	skill_level = SKILL_PILOT_DEFAULT
+	max_skill_level = SKILL_PILOT_MAX
 
 /datum/skill/police
 	skill_name = SKILL_POLICE
+	readable_skill_name = SKILL_POLICE
 	skill_level = SKILL_POLICE_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:139]:
```suggestion
	skill_name = SKILL_POLICE
	skill_level = SKILL_POLICE_DEFAULT
```

File: code/datums/skills.dm [hunk 19]
```diff
@@ -91,47 +117,69 @@
...
 /datum/skill/antag
 	skill_name = SKILL_ANTAG
+	readable_skill_name = "illegal technology"
 	skill_level = SKILL_ANTAG_DEFAULT
+	max_skill_level = SKILL_ANTAG_MAX
 
 /datum/skill/pilot
 	skill_name = SKILL_PILOT
+	readable_skill_name = SKILL_PILOT
 	skill_level = SKILL_PILOT_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:133]:
```suggestion
	skill_name = SKILL_PILOT
	skill_level = SKILL_PILOT_DEFAULT
```

File: code/datums/skills.dm [hunk 20]
```diff
@@ -91,47 +117,69 @@
 
 /datum/skill/research
 	skill_name = SKILL_RESEARCH
+	readable_skill_name = SKILL_RESEARCH
 	skill_level = SKILL_RESEARCH_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:121]:
```suggestion
	skill_name = SKILL_RESEARCH
	skill_level = SKILL_RESEARCH_DEFAULT
```

File: code/datums/skills.dm [hunk 21]
```diff
@@ -64,11 +86,15 @@
...
 /datum/skill/medical
 	skill_name = SKILL_MEDICAL
+	readable_skill_name = SKILL_MEDICAL
 	skill_level = SKILL_MEDICAL_DEFAULT
+	max_skill_level = SKILL_MEDICAL_MAX
 
 /datum/skill/surgery
 	skill_name = SKILL_SURGERY
+	readable_skill_name = SKILL_SURGERY
 	skill_level = SKILL_SURGERY_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:96]:
```suggestion
	skill_name = SKILL_SURGERY
	skill_level = SKILL_SURGERY_DEFAULT
```

File: code/datums/skills.dm [hunk 22]
```diff
@@ -64,11 +86,15 @@
 
 /datum/skill/medical
 	skill_name = SKILL_MEDICAL
+	readable_skill_name = SKILL_MEDICAL
 	skill_level = SKILL_MEDICAL_DEFAULT
```
Comment (harryob) [inline code/datums/skills.dm:90]:
```suggestion
	skill_name = SKILL_MEDICAL
	skill_level = SKILL_MEDICAL_DEFAULT
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**